server/src/main/java/org/elasticsearch/indices/recovery/RemoteRecoveryTargetHandler.java

```diff
@@ -74,25 +91,32 @@ public RemoteRecoveryTargetHandler(long recoveryId, ShardId shardId, TransportSe
                 .withType(TransportRequestOptions.Type.RECOVERY)
                 .withTimeout(recoverySettings.internalActionTimeout())
                 .build();
-
     }
 
     @Override
     public void prepareForTranslogOperations(int totalTranslogOps, ActionListener<Void> listener) {
-        transportService.sendRequest(targetNode, PeerRecoveryTargetService.Actions.PREPARE_TRANSLOG,
-            new RecoveryPrepareForTranslogOperationsRequest(recoveryId, shardId, totalTranslogOps),
-            TransportRequestOptions.builder().withTimeout(recoverySettings.internalActionTimeout()).build(),
-            new ActionListenerResponseHandler<>(ActionListener.map(listener, r -> null),
-                in -> TransportResponse.Empty.INSTANCE, ThreadPool.Names.GENERIC));
+        final TimeValue timeout = recoverySettings.internalActionTimeout();
+        final RecoveryPrepareForTranslogOperationsRequest request =
+            new RecoveryPrepareForTranslogOperationsRequest(recoveryId, shardId, totalTranslogOps);
+        final Consumer<ActionListener<Void>> action = actionListener ->
+            transportService.sendRequest(targetNode, PeerRecoveryTargetService.Actions.PREPARE_TRANSLOG, request,
+                TransportRequestOptions.builder().withTimeout(recoverySettings.internalActionTimeout()).build(),
+                new ActionListenerResponseHandler<>(ActionListener.map(actionListener, r -> null),
+                    in -> TransportResponse.Empty.INSTANCE, ThreadPool.Names.GENERIC));
+        executeRetryableAction(action, listener, timeout);
     }
 
     @Override
     public void finalizeRecovery(final long globalCheckpoint, final long trimAboveSeqNo, final ActionListener<Void> listener) {
-        transportService.sendRequest(targetNode, PeerRecoveryTargetService.Actions.FINALIZE,
-            new RecoveryFinalizeRecoveryRequest(recoveryId, shardId, globalCheckpoint, trimAboveSeqNo),
-            TransportRequestOptions.builder().withTimeout(recoverySettings.internalActionLongTimeout()).build(),
-            new ActionListenerResponseHandler<>(ActionListener.map(listener, r -> null),
-                in -> TransportResponse.Empty.INSTANCE, ThreadPool.Names.GENERIC));
+        final TimeValue timeout = recoverySettings.internalActionTimeout();
```

Comment:
I'm not sure I understand here. `FINALIZE` for example can fail due to circuit breaking. I'm not sure why we would not want to retry? I understand that `FINALIZE` is not resource intensive, but the node can be circuit breaking for other reasons. It seems like we would want to retry everything? Especially since I have reduced the timeout to something reasonable (1 minute).

Reply:
ok